MAINTENANCE - ASCII Module refactoring

carlwilson opened the pull request:
- added markers for all error numbers;
- reviewed and documented errors (in spreadsheet for wiki);
- control characters and line endings handled by dedicated enum types with convenience methods;
- cleaned up reporting of control characters and line endings;
- optimised character handling (IMO);
- removed defunct constants and class members;
- simplified parameter handling;
- using dedicated `ModuleBase.setupDataStream()` method;
- replaced some magic numbers with named constants.
- bumped version and release date (provisional) for ASCII, PDF and BYTESTREAM modules;
- updated test target creation script with new release details for same modules;
- added test target `sed` for ASCII module to change fixed `TAB (0x09)` to `HT (0x09)`; and
- ASCII Module now uses `ModuleBase.setChecksums()` method.
Review after #351

carlwilson committed:
MAINTENANCE - ASCII Module refactoring
- added markers for all error numbers;
- reviewed and documented errors (in spreadsheet for wiki);
- control characters and line endings handled by dedicated enum types with convenience methods;
- cleaned up reporting of control characters and line endings;
- optimised character handling (IMO);
- removed defunct constants and class members;
- simplified parameter handling;
- using dedicated `ModuleBase.setupDataStream()` method; and
- replaced some magic numbers with named constants.

carlwilson committed:
Merge branch 'integration' into refact/ascii-module

carlwilson committed:
FIX - Use ANSI Control Char Mnemonic
- replace `TAB` with official ANSI & ECMA `HT` value; and
- added ANSI name value to `ControlChar`.

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/AsciiModule.java:
```diff
@@ -181,68 +155,61 @@ public final int parse(InputStream stream, RepInfo info, int parseIndex)
-            _dstream = getBufferedDataStream(stream,
-                    _je != null ? _je.getBufferSize() : 0);
-        }
+        setupDataStream(stream, info);
```
[As earlier](https://github.com/openpreserve/jhove/pull/351/files#r189909055).

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/AsciiModule.java:
```diff
@@ -151,25 +136,14 @@ public AsciiModule() {
-            Iterator iter = _defaultParams.iterator();
-            while (iter.hasNext()) {
-                String param = (String) iter.next();
-                if ("withtextmd=true".equalsIgnoreCase(param)) {
```
The new code loses this parameter's case-insensitivity.

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/AsciiModule.java:
```diff
@@ -151,25 +136,14 @@ public AsciiModule() {
-        _controlCharMap = new HashMap<Integer, String>();
+        ControlChar prevChar = null;
+        usedCtrlChars = new HashSet<>();
+        usedLineEndings = new HashSet<>();
```
I just learnt about EnumSets. EnumSets sound cool. That is all.

david-russo reviewed (commented):
Enums make everything better.

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/ascii/ControlChar.java:
```diff
@@ -47,14 +48,17 @@
+	public final String ansiName;
 	/**
-	 * JHOVE's reporting mnemonic, the code followed by the int value in hex.
+	 * JHOVE's reporting mnemonic, the code followed by the in value in hex.
```
Overzealous auto-correct?

carlwilson committed:
Merge remote-tracking branch 'origin/integration' into refact/ascii-module

carlwilson committed:
REVIEW - Addressing review comments + more
- bumped version and release date (provisional) for ASCII, PDF and BYTESTREAM modules;
- upated test target creation script with new release details for same modules;
- addressed case-insensitivity for param check, via a dedicated `ModuleBase.isParamInDefaults()` method;
- added test target `sed` for ASCII module to change fixed `TAB (0x09)` to `HT (0x09)`;
- ASCII Module now uses `ModuleBase.setChecksums()` method; and
- addressed commenting review comment.

carlwilson committed:
Merge branch 'integration' into refact/ascii-module

carlwilson committed:
Merge remote-tracking branch 'origin/integration' into refact/ascii-module
